src/events/processCollectedSignatures/estimateGas.js

```diff
@@ -0,0 +1,51 @@
+const { HttpListProviderError } = require('http-list-provider')
+const {
+  AlreadyProcessedError,
+  IncompatibleContractError,
+  InvalidValidatorError
+} = require('../../utils/errors')
+const { parseMessage } = require('../../utils/message')
+
+async function estimateGas({
+  foreignBridge,
+  validatorContract,
+  message,
+  numberOfCollectedSignatures,
+  v,
+  r,
+  s,
+  address
+}) {
+  try {
+    const gasEstimate = await foreignBridge.methods
+      .executeSignatures(v, r, s, message)
+      .estimateGas()
+    return gasEstimate
+  } catch (e) {
+    if (e instanceof HttpListProviderError) {
+      throw e
+    }
+
+    const requiredSignatures = await foreignBridge.methods.requiredSignatures().call()
+    if (requiredSignatures.toString() !== numberOfCollectedSignatures.toString()) {
```

Comment:
Number of collected signatures could be more than required. It is OK.
Here is the corresponding check from contracts:
```solidity
        uint256 requiredSignatures = _validatorContract.requiredSignatures();
        require(_vs.length >= requiredSignatures);
```
And I'd move this check after the check that the message already relayed because the second one is more probable.

Reply:
Done.